stories/spectra/nasqad.stories.tsx

```diff
@@ -0,0 +1,91 @@
+import { Meta } from '@storybook/react';
+import React, { useState, useEffect } from 'react';
+
+import { Axis, LineSeries, Plot, Heading } from '../../src';
+import type { Series } from '../../src/types';
+import srcData from '../data/nasdaq.json';
+
+export default {
+  title: 'Experimental spectra/Nasqad',
+  argTypes: {
+    refreshFrequency: { defaultValue: 1, control: 'number' },
+    step: { defaultValue: 1, control: 'number' },
+    displayInterval: { defaultValue: 500, control: 'number' },
+  },
+} as Meta;
+
+/** Util functions *****************/
+
+const generateNewXY = (serie: Series, step: number): [number, number] => {
+  const x = serie.x + step;
+  const rand = Math.floor(Math.random() * 10);
+  const y = rand % 2 === 0 ? serie.y + rand * 0.1 : serie.y - rand * 0.1; // generate a new Y by + or - a random value to the last point.y
+  return [x, y];
+};
+
+const getLastData = (
+  data: Array<Series>,
+  displayInterval: number,
+): Array<Series> => {
+  const lastTimestamp = data[data.length - 1].x;
+  const firstIndex = data.findIndex(
+    (serie) => serie.x >= lastTimestamp - displayInterval,
+  );
+  const newVals = data.slice(firstIndex);
+  return newVals;
+};
+
+/** Main Component *****************/
+
+type Props = Record<string, number>;
+export function NasqadExample(props: Props) {
+  const {
+    refreshFrequency, // Page refresher frequency (s)
+    step, // timestamp's step
+    displayInterval, // interval to display (s)
+  } = props;
+
+  function PlotRefresher() {
+    const [data, setData] = useState(srcData);
+
+    useEffect(() => {
+      const timer = setInterval(() => {
```

Comment:
In this case you could use setTimeout (and cancel the timeout in the return). Anyway it will never be called more than once (if I understand well).

Reply:
Exactly,  
setTimeout will never be called more than once, which is not what we aim for.
setInterval allows us to run a function repeatedly, starting after the interval of time, then repeating continuously at that interval.

Reply:
Yes, setInterval allows us to run the function repeatedly, but this is not what happens here, because everytime the `setData` function is called, the component is re-rendered, the `useEffect` cleanup function is executed, clearing the interval and a new interval is created. So in practice, setTimeout has the same effect in this situation.

Reply:
You are right, 
setTimeout is a better fit since the hook cleanup the timer after every render